src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -96,22 +96,24 @@ bool is_output_buffer(const primitive_inst* prim, bool runtime_alloc) {
 }
 
 bool is_user_cpu(const program_node* user) {
+    // If the user is dynamic and runtime skippable node, we still need to its parents' completion event
+    // If the user is not runtime skippable in dynamic, we need to check it is cpu impl
+    if (user->is_dynamic() && user->is_runtime_skippable())
+        return true;
```

Comment:
Umm... this is too conservative. This is setting compeletion event even though the user is not cpu. 
e.g., gpu_kernel1->reorder(can_be_opt)->gpu_kernel2 // here gpu_kernel1 is always setting compeletion event. 

I think we should do like 

bool is_user_cpu(const program_node* user) {
    if (user->can_be_optimized() || user->is_runtime_skippable()) {
        auto users = user->get_users();
        for (const auto& u : users) {
            if (is_user_cpu(u)) {
                return true;
            }
        }
    }
    bool is_cpu = user->get_selected_impl() ? user->get_selected_impl()->is_cpu()
                                            : user->get_preferred_impl_type() == impl_types::cpu;
}

How do you think?

Reply:
Done.